src/sentry/static/sentry/app/views/performance/table.tsx

```diff
@@ -0,0 +1,369 @@
+
+    const lastIndex = columnOrder.length - 1;
+    return columnOrder.map((column, index) => {
+      const field = String(column.key);
+      const fieldName = getAggregateAlias(field);
+      const fieldType = tableMeta[fieldName];
+
+      const fieldRenderer = getFieldRenderer(field, tableMeta);
+      let rendered = fieldRenderer(row, {organization, location});
+
+      const isFirstCell = index === 0;
+      const isLastCell = index === lastIndex;
+
+      if (isFirstCell) {
+        // the first column of the row should link to the transaction details view
+        // on Discover
+
+        const eventSlug = generateEventSlug(row);
+
+        const target = eventDetailsRouteWithEventView({
+          orgSlug: organization.slug,
+          eventSlug,
+          eventView,
+        });
+
+        rendered = <Link to={target}>{rendered}</Link>;
+      }
+
+      const isNumeric = ['integer', 'number', 'duration'].includes(fieldType);
+      if (isNumeric) {
+        return (
+          <BodyCell
+            key={column.key}
+            first={isFirstCell}
+            last={isLastCell}
+            hideBottomBorder={isLastRow}
+          >
+            <NumericColumn>{rendered}</NumericColumn>
+          </BodyCell>
+        );
+      }
+
+      return (
+        <BodyCell
+          key={column.key}
+          first={isFirstCell}
+          last={isLastCell}
+          hideBottomBorder={isLastRow}
+        >
+          {rendered}
+        </BodyCell>
+      );
+    });
+  };
+
+  generateSortLink = (field: Field, tableDataMeta?: MetaType) => ():
+    | LocationDescriptorObject
+    | undefined => {
+    const {eventView} = this.props;
+
+    if (!tableDataMeta) {
+      return undefined;
+    }
+
+    const nextEventView = eventView.sortOnField(field, tableDataMeta);
+    const queryStringObject = nextEventView.generateQueryStringObject();
+
+    const omitKeys = ['widths', 'query', 'name', 'field'];
+
+    return {
+      ...location,
+      query: omit(queryStringObject, omitKeys),
+    };
+  };
+
+  renderHeader = () => {
+    const {location, eventView} = this.props;
+    const {tableData} = this.state;
+
+    const columnOrder = this.props.eventView.getColumns();
+
+    const lastindex = columnOrder.length - 1;
+    return columnOrder.map((column, index) => {
+      const field = column.eventViewField;
+
+      // establish alignment based on the type
+      const alignedTypes: ColumnValueType[] = ['number', 'duration', 'integer'];
+      let align: Alignments = alignedTypes.includes(column.type) ? 'right' : 'left';
+
+      if (column.type === 'never' || column.type === '*') {
+        // fallback to align the column based on the table metadata
+        const maybeType =
+          tableData && tableData.meta
+            ? tableData.meta[getAggregateAlias(field.field)]
+            : undefined;
+
+        if (maybeType === 'integer' || maybeType === 'number') {
+          align = 'right';
+        }
```

Comment:
If you set the alignment after inferring the type you wouldn't need this code.

Reply:
This came from the table implementation in Eventsv2 

I've consolidated them into a new component called `<HeaderCell />` so that it will be easier to refactor later on https://github.com/getsentry/sentry/pull/16906/commits/8062cf27f1a8b965cf31c113c7ec35786c5f62be